packages/webapp/src/components/Animals/DetailCards/General.tsx

```diff
@@ -40,10 +40,29 @@ import {
   Option as AnimalSelectOption,
   AnimalBreedSelect,
 } from '../AddAnimalsFormCard/AnimalSelect';
+import { generateUniqueAnimalId, parseUniqueDefaultId } from '../../../util/animal';
+
+// Add Animals Flow
+type SingleAnimalTypeUses = Option[DetailsFields.USE][];
+
+// Single Animal View
+type AnimalTypeToUsesMapping = {
+  default_type_id: number;
+  uses: Option[DetailsFields.USE][];
+}[];
+
+type UseOptions = SingleAnimalTypeUses | AnimalTypeToUsesMapping;
```

Comment:
I think the type should be either of them. Can we remove `useOptionsForType` (filtered uses) from `AddAnimalDetails`, so the filtering always happens in this component?

Reply:
Agreed!